train.py

```diff
@@ -16,65 +17,60 @@ def train(data_cases):
 def recursive(data_cases):
```

Comment:
Could make this function name more descriptive. Here's a suggestion...you could see how it reads with the function names in a line that calls it:
`model.treeRoot=buildModelTreeRecursive(data_cases)`
vs.
`model.treeRoot=recursive(data_cases)`
And then read it out loud. If it sounds good, that's a way to decide if you or somebody else in the future can determine what the code is supposed to do.